dspace-server-webapp/src/test/java/org/dspace/app/rest/RequestItemRepositoryIT.java

```diff
@@ -0,0 +1,368 @@
+    }
+
+    /**
+     * Test of findOne method, with an authenticated user.
+     *
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testFindOneAuthenticated()
+            throws Exception {
+        System.out.println("findOne");
+
+        context.turnOffAuthorisationSystem();
+
+        // Create necessary supporting objects.
+        Collection collection = CollectionBuilder.createCollection(context, parentCommunity)
+                .build();
+        Item item = ItemBuilder.createItem(context, collection)
+                .build();
+        InputStream is = new ByteArrayInputStream(new byte[0]);
+        Bitstream bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                .build();
+
+        // Create a request.
+        RequestItem request = RequestItemBuilder
+                .createRequestItem(context, item, bitstream)
+                .build();
+
+        // Test:  can we find it?
+        String authToken = getAuthToken(admin.getEmail(), password);
+        final String uri = URI_ROOT + '/' + request.getToken();
+        getClient(authToken).perform(get(uri))
+                   .andExpect(status().isOk()) // Can we find it?
+                   .andExpect(content().contentType(contentType))
+                   .andExpect(jsonPath("$", Matchers.is(
+                       RequestCopyMatcher.matchRequestCopy(request))));
+
+        // Clean up.
+        bitstream.setDeleted(true);
+        context.restoreAuthSystemState();
+    }
+
+    /**
+     * Test of findOne method, with an UNauthenticated user.
+     *
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testFindOneNotAuthenticated()
+            throws Exception {
+        System.out.println("findOne");
+
+        context.turnOffAuthorisationSystem();
+
+        // Create necessary supporting objects.
+        Collection collection = CollectionBuilder.createCollection(context, parentCommunity)
+                .build();
+        Item item = ItemBuilder.createItem(context, collection)
+                .build();
+        InputStream is = new ByteArrayInputStream(new byte[0]);
+        Bitstream bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                .build();
+
+        // Create a request.
+        RequestItem request = RequestItemBuilder
+                .createRequestItem(context, item, bitstream)
+                .build();
+
+        // Test:  can we find it?
+        final String uri = URI_ROOT + '/' + request.getToken();
+        getClient().perform(get(uri))
+                   .andExpect(status().isOk()) // Can we find it?
+                   .andExpect(content().contentType(contentType))
+                   .andExpect(jsonPath("$", Matchers.is(
+                       RequestCopyMatcher.matchRequestCopy(request))));
+
+        // Clean up.
+        bitstream.setDeleted(true);
+        context.restoreAuthSystemState();
```

Comment:
Same as previous.  Auth needs to be turned back on sooner.  Cleanup likely isn't necessary.

Reply:
Done.